ml-agents/mlagents/trainers/learn.py

```diff
@@ -395,14 +396,12 @@ def create_environment_factory(
     env_args: Optional[List[str]],
 ) -> Callable[[int, List[SideChannel]], BaseEnv]:
     if env_path is not None:
-        # Strip out executable extensions if passed
-        env_path = (
-            env_path.strip()
-            .replace(".app", "")
-            .replace(".exe", "")
-            .replace(".x86_64", "")
-            .replace(".x86", "")
-        )
+        launch_string = UnityEnvironment.validate_environment_path(env_path)
+        if launch_string is None:
+            raise UnityEnvironmentException(
+                "Couldn't launch the {0} environment. "
```

Comment:
Small nit: I'd like to move away from `str.format()` like this for new code, and use [f-strings](https://realpython.com/python-f-strings/) when possible. This could be written instead as
```python
raise UnityEnvironmentException(
    f"Couldn't launch the {env_path} environment. Provided filename does not match any environments."
)
```
We're not enforcing this, so feel free to ignore.